pyhap/characteristic.py

```diff
@@ -177,19 +208,20 @@ def set_value(self, value, should_notify=True):
         """
         logger.debug('%s: Set value to %s', self.display_name, value)
         value = self.to_valid_value(value)
-        self.value = value
+        self._value = value
         if should_notify and self.broker:
             self.notify()
 
     def client_update_value(self, value):
         """Called from broker for value change in Home app.
 
-        Change self.value to value and call callback.
+        Call set_value and call callback.
         """
         logger.debug('%s: Client update value to %s',
-                      self.display_name, value)
-        self.value = value
-        self.notify()
+                     self.display_name, value)
+        # Set new value before anything else, set_value calls notify
+        self.set_value(value)
+        # Call setter_callback
```

Comment:
The same here. It would be quite confusion if a `client_update_value` call cause a `set_value` log output as well.
Use `self._value = value` instead.